pkg/media/rtp/jitter.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rtp
+
+import (
+	"time"
+
+	"github.com/livekit/server-sdk-go/v2/pkg/jitter"
+	"github.com/pion/rtp"
+)
+
+const (
+	jitterMaxLatency = 2 * time.Second
+)
+
+func HandleJitter(clockRate int, h Handler) Handler {
+	return &jitterHandler{
+		h:   h,
+		buf: jitter.NewBuffer(audioDepacketizer{}, uint32(clockRate), jitterMaxLatency),
```

Comment:
This will add up to a 2 second delay to the SIP stream. Is this the behavior we want?

Reply:
how is jitter handled on the clients? we might still want it, but with a much lower max latency than 2s

Reply:
Also wondering what the mixer is going to do. Is it going wait for the missing track, or will it mix whatever tracks it has before that?

Reply:
The mixer will still buffer ~60 ms of audio.

If because of a jitter these packets are delayed more than 60 ms, mixer will at first use the buffered frames. As soon as it reaches 0 buffer on that track, mixer will mute it track and will mix the rest. As soon as 60 ms is available on that track again, it will be unmuted and mixed as before.

Reply:
Does this means that if the jitter buffer delays samples by more than 60ms, the participant will be muted until late packets arrive, and then speech will restart, but be mixed in late, up to 500ms, until the participant leaves?

Reply:
Right, if the jitter buffer delays packets up to 500 ms, the participant will be muted during this time, but then only last 100 ms of that track will resume and this delay will persist. The 60 ms I mentioned is the target latency, while 100 ms is the max buffer.

Actually, it will almost certainly drop from 100 ms to at least 80 ms because some packets might arrive in a burst and at least one will go over the max buffer, causing drop of an older frame.

Ideally mixer should buffer all 500 ms for a track, delay it after resume, but also resample so that it can eventually catch up. But this is a bit too complicated for now.

We could increase the mixer buffer up to ~250 ms, and thus let it drop half of the frames in case of extreme jitter. But still, this delay will persist until the participant leaves, so it might not be acceptable.

Reply:
I just think that we should make the jitter buffer and mixer delay consistent as there is no point waiting for data we'll drop in the pipeline.

Reply:
In that case it should be 60 ms I think.